rpc-tests/tests/test-rpc.js

```diff
@@ -21,69 +25,80 @@ describeWithNetwork(network, `${network} RPC`, function(context) {
 		expect(chain.toString()).to.equal(`${capitalize(network)} Testnet`);
 	});
 
-	it('should fetch chain name from rpc node', async function () {
+	it('should fetch chain name from rpc node', async () => {
 		const name = await context.api.rpc.system.name();
 
 		expect(name.toString()).to.equal('Astar Collator');
 	});
 
-	it('should be able to Register contract on H160 address 0x01 using Alice account', async function () {
-        const api = context.api;
-
-        // Construct the keyring after the API (crypto has an async init)
-        const keyring = new Keyring({ type: 'sr25519' });
-      
-        // Add Alice to our keyring with a hard-derivation path (empty phrase, so uses dev)
-        const alice = keyring.addFromUri('//Alice');
-
-        // Create a extrinsic, transferring 100 units to Bob
-        const transfer = api.tx.dappsStaking.register(getAddressEnum(CONTRACT));
-      
-        // Sign and send the transaction using our account
-        const hash = await transfer.signAndSend(alice, { nonce: -1 });
-
-        console.log('Transfer sent with hash', hash.toHex());
-
-        expect(hash).to.exist;
+	it('should be able to Register contract on H160 address 0x01 using Alice account', async () => {
+        const events = await sendTransaction(
+            context.api,
+            context.api.tx.dappsStaking.register(getAddressEnum(CONTRACT)),
+            context.alice,
+            'dappsStaking',
+            'NewContract'
+        );
+
+        expect(events.length).to.equal(1);
+        expect(events[0].event.section).to.equal('dappsStaking');
+        expect(events[0].event.method).to.equal('NewContract');
+        expect(events[0].event.data[0].toString()).to.equal(ALICE);
```

Comment:
A better approach would be to check on-chain state before and after.
You expect that there is a registered contract with the specified address after register has successfully been executed, right?

Reply:
Done